nebula/ngData/data.py

```diff
@@ -82,33 +96,40 @@ def decode_value(self, value, schema_version=None):
         if ver_bytes_num > 0:
             for i in range(ver_bytes_num):
                 if six.PY2:
-                    ver |= ord(value[self._offset]) << 8
+                    ver |= ord(value[self._offset])
                 else:
-                    ver |= value[self._offset] << 8
+                    ver |= value[self._offset]
                 self._offset += 1
-        if ver != schema_version:
-            raise Exception('parsed version %d is not equal to version %d provided', ver, schema_version)
-        self._offset += blk_off_bytes_num * (self._field_num // 16)
-        properties = []
-        for i in range(len(self._defs)):
-            field = self._defs[i][0]
-            property_type = self._defs[i][1]
+        self._offset += blk_off_bytes_num * (self._field_num[ver] // 16)
+        properties = {}
+        for i in range(len(self._defs[ver])):
+            field = self._defs[ver][i][0]
+            property_type = self._defs[ver][i][1]
             if property_type == SupportedType.BOOL:
-                properties.append(self.get_bool_property(field, value))
+                properties[field] = self.get_bool_property(field, value)
             elif property_type == SupportedType.INT:
-                properties.append(self.get_int_property(field, value))
+                properties[field] = self.get_int_property(field, value)
             elif property_type == SupportedType.FLOAT: #unused now
-                properties.append(self.get_float_property(field, value))
+                properties[field] = self.get_float_property(field, value)
             elif property_type == SupportedType.DOUBLE:
-                properties.append(self.get_double_property(field, value))
+                properties[field] = self.get_double_property(field, value)
             elif property_type == SupportedType.STRING:
-                properties.append(self.get_string_property(field, value))
+                properties[field] = self.get_string_property(field, value)
             elif property_type == SupportedType.TIMESTAMP:
-                properties.append(self.get_timestamp_property(field, value))
+                properties[field] = self.get_timestamp_property(field, value)
             else:
                 raise Exception('Unsupported propertyType in schema: ', property_type)
 
-        return properties
+        returned_props = []
+        for defs in self._defs[self._version]:
+            prop_name = defs[0]
+            prop_type = defs[1]
+            if prop_name in properties.keys():
+                returned_props.append(properties[prop_name])
```

Comment:
if the user changes the column's type, the origin value and new type do not match.

Reply:
Thank you, I overlooked that